### PR TITLE
Add code coverage for ToolStripLabel.cs file

### PR DESCRIPTION
**Pull request by Zheng-Li01**
related https://github.com/dotnet/winforms/issues/10453

Proposed changes
Add unit tests for ToolStripLabe.cs to test its properties & methods & events
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12166)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    [WinFormsFact]
+    public void ToolStripLabel_ShouldSerializeActiveLinkColor_ReturnsFalse_WhenActiveLinkColorIsEmpty()
+    {
+        _toolStripLabel.ActiveLinkColor = Color.Empty;
+
+        var accessor = _toolStripLabel.TestAccessor().Dynamic;
+        bool result = accessor.ShouldSerializeActiveLinkColor();
+
+        result.Should().BeFalse();
+    }
```
Comment (Epica3055) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:285]: And this two too.
And cases below as well.

**Review by ricardobossan — COMMENTED** [review]
Other than small comments, LGTM

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 2]
```diff
@@ -9,14 +9,28 @@ namespace System.Windows.Forms.Tests;
...
+    public void ToolStripLabel_DefaultConstructor_SetsDefaults()
     {
-        _toolStripLabel = new();
+        _toolStripLabel.Text.Should().BeEmpty();
+        _toolStripLabel.Image.Should().BeNull();
+        _toolStripLabel.IsLink.Should().BeFalse();
     }
 
-    public void Dispose() => _toolStripLabel.Dispose();
+    [WinFormsFact]
```
Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:24]: In every test method that asserts constructor behavior, the constructor should be called at the `Act` step of the method, like this:

```cs
  [WinFormsFact]
  public void ToolStripLabel_ConstructorWithText_SetsText()
  {
      string text = "Test Label";
      
      // Act
      ToolStripLabel toolStripLabel = new(text);

      toolStripLabel.Text.Should().Be(text);
  }

  [WinFormsFact]
  public void ToolStripLabel_ConstructorWithImage_SetsImage()
  {
      using Bitmap image = new(10, 10);
      
      // Act
      ToolStripLabel toolStripLabel = new(image);

      toolStripLabel.Image.Should().Be(image);
  }
  ```
  
 Other methods in this PR code also need that adjustment.

**Review by ricardobossan — APPROVED** [review]

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+public class ToolStripLabelTests : IDisposable
+{
+    private readonly ToolStripLabel _toolStripLabel = new();
+    public void Dispose() => _toolStripLabel.Dispose();
+
+    [WinFormsFact]
+    public void ToolStripLabel_DefaultConstructor_SetsDefaults()
+    {
+        ToolStripLabel toolStripLabel = new();
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:18]: Please dispose these
```suggestion
        using ToolStripLabel toolStripLabel = new();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        toolStripLabel.Image.Should().BeNull();
+        toolStripLabel.IsLink.Should().BeFalse();
+    }
+
+    [WinFormsFact]
+    public void ToolStripLabel_ConstructorWithText_SetsText()
+    {
+        string text = "Test Label";
+
+        ToolStripLabel toolStripLabel = new(text);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:30]:
```suggestion
        using ToolStripLabel toolStripLabel = new(text);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+        toolStripLabel.Image.Should().BeNull();
+        toolStripLabel.IsLink.Should().BeFalse();
+    }
+
+    [WinFormsFact]
+    public void ToolStripLabel_ConstructorWithImage_SetsImage()
+    {
+        using Bitmap image = new(10, 10);
+
+        ToolStripLabel toolStripLabel = new(image);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:42]:
```suggestion
        using ToolStripLabel toolStripLabel = new(image);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+        toolStripLabel.IsLink.Should().BeFalse();
+    }
+
+    [WinFormsFact]
+    public void ToolStripLabel_ConstructorWithTextAndImage_SetsTextAndImage()
+    {
+        using Bitmap image = new(10, 10);
+        string text = "Test Label";
+
+        ToolStripLabel toolStripLabel = new(text, image);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:55]:
```suggestion
        using ToolStripLabel toolStripLabel = new(text, image);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    [WinFormsFact]
+    public void ToolStripLabel_ConstructorWithTextImageAndIsLink_SetsTextImageAndIsLink()
+    {
+        using Bitmap image = new(10, 10);
+        string text = "Test Label";
+        bool isLink = true;
+
+        ToolStripLabel toolStripLabel = new(text, image, isLink);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:69]:
```suggestion
        using ToolStripLabel toolStripLabel = new(text, image, isLink);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+    [WinFormsFact]
+    public void ToolStripLabel_ConstructorWithTextImageIsLinkAndOnClick_SetsTextImageIsLinkAndOnClick()
+    {
+        using Bitmap image = new(10, 10);
+        string text = "Test Label";
+        bool isLink = true;
+        bool eventHandlerCalled = false;
+        EventHandler onClick = (sender, e) => eventHandlerCalled = true;
+
+        ToolStripLabel toolStripLabel = new(text, image, isLink, onClick);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:85]:
```suggestion
        using ToolStripLabel toolStripLabel = new(text, image, isLink, onClick);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+        _toolStripLabel.LinkColor = expectedColor;
+
+        _toolStripLabel.LinkColor.Should().Be(expectedColor);
+    }
+
+    [WinFormsFact]
+    public void ToolStripLabel_InvalidateLinkFonts_DisposesFonts()
+    {
+        var accessor = _toolStripLabel.TestAccessor().Dynamic;
+        accessor._linkFont = new Font("Arial", 10);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:172]: Do these Fonts need to be disposed?

Comment (Zheng-Li01) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripLabelTests.cs:172]: In the context of the ToolStripLabelTests class, the **InvalidateLinkFonts** method is responsible for disposing of these fonts. https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStripLabel.cs

   ```
private void InvalidateLinkFonts()
   {
       _linkFont?.Dispose();

       if (_hoverLinkFont is not null && _hoverLinkFont != _linkFont)
       {
           _hoverLinkFont.Dispose();
       }

       _linkFont = null;
       _hoverLinkFont = null;
   }
```

**Review by lonitra — APPROVED** [review]
Thank you

**Pull request merged**